### ISSUE TITLE
pg_repack causes PostgreSQL process crash with pg_statsinfo 3.0 on PostgreSQL 9.4.

### ISSUE THREAD

**Issue by bwtakacy** [reorg/pg_repack#43]
We, pg_reorg team encountered this trouble and found pg_repack has the same one.
This is not a serious problem for users who do not use pg_statsinfo, but the root cause of this problem can be happened in using pg_repack and other PostgreSQL extensions together.
I feel that it is safe to do similar fix in pg_repack.

Here is a detail report.

## Environment
・PostgreSQL 9.4
・pg_statsinfo 3.0.2
・pg_reorg 1.1.11 or pg_repack 1.3.1

## How to reproduce
Start PostgreSQL with setting shared_preload_libraries = 'pg_statsinfo'.

$ psql -d postgres -c "CREATE TABLE test1 ( a int PRIMARY KEY);"
$ pg_reorg -d postgres -t test1 -o a
or
$ pg_repack -d postgres -t test1 -o a

Then, postgres backend process cranshes with segmentation fault.

## Cause
The segmentation fault is happened in appendStringInfoVA_s() function defined in pg_statsinfo.
Though the function with the same name exists in pg_reorg/pg_repack and this does not cause problem, the postgres backend process choose the one in pg_statsinfo side and this is not compatible with PostgreSQL 9.4.

We found, if there are several function definitions with the same name in different shared libraries, 
the postgres process uses the one defined in the shared library loaded first. 

## Fix
We decided that it is better not to export internal functions from reorg shared library in order to avoid symbol conflicts. This can be done by using linker option --version-script and we have found that libpq does this; use SHLIB_EXPORTS environment variable in Makefile and add text file where the functions to be exported are listed.
The below is the commit for pg_reorg.
https://github.com/bwtakacy/pg_reorg/commit/5d9bbaeee0d8a1a8aceec3f35275bd6c55eab606

## Reference
* Controlling symbol visibility for shared libraries: Part 1 - Introduction to symbol visibility
https://www.ibm.com/developerworks/aix/library/au-aix-symbol-visibility/

* symbol confilects in oracle_fdw and OpenLDAP
Using RTLD_DEEPBIND to handle symbol conflicts in loaded libraries http://www.postgresql.org/message-id/CA+CSw_tPDYgnzCYW0S4oU0mTUoUhZ9pc7MRBPXVD-3Zbiwni9w@mail.gmail.com

* symbol confilicts in postgres_fdw and pg_shard
postgres_fdw does not work when pg_shard is on shared_preload_libraries
https://github.com/citusdata/pg_shard/issues/96

Regards,
Takashi Ohnishi

**Comment by schmiddy** [comment]
Thanks for the report and the recommended fix. Hopefully this should be fixed now in pg_repack with 949c0ad150, at least I was able to verify that pg_repack now seems to behave with pg_statsinfo loaded on my machine.

I was curious about verifying that the new symbol table for lib/pg_repack.so was changed as-intended. In the [article from IBM you linked to](https://www.ibm.com/developerworks/aix/library/au-aix-symbol-visibility/), they discussed viewing the symbol table using a tool called `dump`, and the equivalent on Linux seems to be  `nm -D`. Here's the output from [before the fix](https://gist.github.com/schmiddy/e777a55b311ac21ca5ec) and [after the fix](https://gist.github.com/schmiddy/848ebb980d1238940c33). I see a few symbol changes which could plausibly be helping, such as not exporting `execute_with_args` and friends.
